sumpy/expansion/local.py

```diff
@@ -676,12 +926,17 @@ def translate_from(self, src_expansion, src_coeff_exprs, src_rscale,
 
 
 class H2DLocalExpansion(_FourierBesselLocalExpansion):
-    def __init__(self, kernel, order, use_rscale=None):
+    def __init__(self, kernel, order, use_rscale=None,
+            use_preprocessing_for_m2l=False):
         from sumpy.kernel import HelmholtzKernel
         assert (isinstance(kernel.get_base_kernel(), HelmholtzKernel)
                 and kernel.dim == 2)
 
-        super().__init__(kernel, order, use_rscale)
+        super().__init__(kernel, order, use_rscale,
+                use_preprocessing_for_m2l=use_preprocessing_for_m2l)
+
+        if use_preprocessing_for_m2l:
+            raise ValueError("H2DLocalExpansion with FFT is not implemented yet.")
```

Comment:
* Comment on the state here. (You mention that sympy certifies equality, but numerically "something is up". Also, potential `rscale` issues.)
* Move exception to base class?

Reply:
Fixed in b5cdb037bd659c9f26961aa7d9f7434ddd4a82d0